src/data/random/financial.js

```diff
@@ -1,155 +1,127 @@
 export default function() {
-
-    var mu = 0.1,
-        sigma = 0.1,
-        startPrice = 100,
-        startVolume = 100000,
-        startDate = new Date(),
-        stepsPerDay = 50,
-        volumeNoiseFactor = 0.3,
-        filter = function(d) { return true; };
-
-    var calculateOHLC = function(days, prices, volumes) {
-
-        var ohlcv = [],
-            daySteps,
-            currentStep = 0,
-            currentIntraStep = 0;
-
-        while (ohlcv.length < days) {
-            daySteps = prices.slice(currentIntraStep, currentIntraStep + stepsPerDay);
-            ohlcv.push({
-                date: new Date(startDate.getTime()),
-                open: daySteps[0],
-                high: Math.max.apply({}, daySteps),
-                low: Math.min.apply({}, daySteps),
-                close: daySteps[stepsPerDay - 1],
-                volume: volumes[currentStep]
-            });
-            currentIntraStep += stepsPerDay;
-            currentStep += 1;
-            startDate.setUTCDate(startDate.getUTCDate() + 1);
-        }
-        return ohlcv;
+    var startDate = new Date();
+    var startPrice = 100;
+    var granularity = 60 * 60 * 24;
+    var filter = null;
+    var volume = function() {
+        var randomNormal = d3.random.normal(1, 0.1);
+        return Math.ceil(randomNormal() * granularity);
     };
+    var walk = _walk();
 
-    function calculateInterval(start, days) {
+    function granularityInYears() {
         var millisecondsPerYear = 3.15569e10;
-
-        var toDate = new Date(start.getTime());
-        toDate.setUTCDate(start.getUTCDate() + days);
-
-        return {
-            toDate: toDate,
-            years: (toDate.getTime() - start.getTime()) / millisecondsPerYear
-        };
+        return (granularity * 1000) / millisecondsPerYear;
     }
 
-    function dataGenerator(days, years) {
-
-        var prices = walk()
-            .period(years)
-            .steps(days * stepsPerDay)
-            .mu(mu)
-            .sigma(sigma)(startPrice);
-
-        var volumes = walk()
-            .period(years)
-            .steps(days)
-            .mu(0)
-            .sigma(sigma)(startVolume);
-
-        // Add random noise
-        volumes = volumes.map(function(vol) {
-            var boundedNoiseFactor = Math.min(0, Math.max(volumeNoiseFactor, 1));
-            var multiplier = 1 + (boundedNoiseFactor * (1 - 2 * Math.random()));
-            return Math.floor(vol * multiplier);
-        });
-
-        // Save the new start values
-        startPrice = prices[prices.length - 1];
-        startVolume = volumes[volumes.length - 1];
-
-        return calculateOHLC(days, prices, volumes).filter(function(d) {
-            return filter(d.date);
-        });
+    function calculateOHLC(start, price) {
+        var prices = walk
+            .period(granularityInYears())(price);
+        var datum = {
+            date: start,
+            open: prices[0],
+            high: Math.max.apply({}, prices),
```

Comment:
I think the context argument should be ```Math```? E.g. this is what Babel produces for spread operators -
```js
Math.max(...[1,2,3])
/// --->
Math.max.apply(Math, [1, 2, 3]);
```